addons/mail/static/src/core/web_portal/message_patch.js

```diff
@@ -125,16 +125,28 @@ patch(Message.prototype, {
                 this.insertEllipsisbtn(childEl);
             }
         }
-
+        let buttonStyles;
+        let iconStyles;
+        const buttons = [];
+        const icons = [];
         for (const group of groups) {
             const index = this.state.lastReadMoreIndex++;
             const ellipsisbtnEl = document.createElement("button");
-            ellipsisbtnEl.style.display = "block";
-            ellipsisbtnEl.style.marginLeft = "-3px";
-            ellipsisbtnEl.className = "o-mail-ellipsis badge rounded-pill border-0 py-0 px-1";
+            ellipsisbtnEl.className = "o-mail-ellipsis d-block badge rounded-pill border-0 py-0 px-1";
+            ellipsisbtnEl.style.visibility = "hidden";
             const iconellipsisEl = document.createElement("i");
-            iconellipsisEl.className = "fa fa-ellipsis-h fa-lg";
+            iconellipsisEl.innerText = "⋯";
+            iconellipsisEl.style.userSelect = "none";
+            iconellipsisEl.style.visibility = "hidden";
             ellipsisbtnEl.append(iconellipsisEl);
+            document.body.appendChild(ellipsisbtnEl);
+            buttonStyles = {};
+            Object.assign(buttonStyles, document.defaultView.getComputedStyle(ellipsisbtnEl));
+            iconStyles = {};
+            Object.assign(iconStyles, document.defaultView.getComputedStyle(iconellipsisEl));
+            ellipsisbtnEl.remove();
+            buttons.push(ellipsisbtnEl);
+            icons.push(iconellipsisEl);
```

Comment:
Couldn't you rely on a template instead of crafting element by hand in JS, i.e. with `renderToElement()`?
Even doing parts in templates and reducing building stuffs in JS, this would reduce the LOCs and scrolling to guess and understand this code. Each template part would also be named in a way to give a better grasp of the resulting changes in DOM.